core/include/drivers/gpio.h

```diff
@@ -166,16 +199,39 @@ TEE_Result gpio_dt_alloc_pin(struct dt_pargs *pargs, struct gpio **gpio);
 TEE_Result gpio_dt_get_by_index(const void *fdt, int nodeoffset,
 				unsigned int index, const char *gpio_name,
 				struct gpio **gpio);
+
+/**
+ * gpio_dt_cfg_by_index() - Get a GPIO controller at a specific index in
+ * 'gpios' property and configure with provided flags
+ *
+ * @fdt: Device tree to work on
+ * @nodeoffset: Node offset of the subnode containing a 'gpios' property
+ * @index: GPIO pin index in '*-gpios' property
+ * @gpio_name: Prefix of the GPIO properties in device tree, can be NULL
+ * @flags: requester flags of GPIO
+ * @gpio: Output GPIO pin reference upon success
+ *
+ * Return TEE_SUCCESS in case of success
+ * Return TEE_ERROR_DEFER_DRIVER_INIT if GPIO controller is not initialized
+ * Return a TEE_Result compliant code in case of error
+ */
+TEE_Result gpio_dt_cfg_by_index(const void *fdt, int nodeoffset,
+				unsigned int index, const char *gpio_name,
+				enum gpio_flags mode,
+				struct gpio **gpio);
 #else
-static inline TEE_Result gpio_dt_get_by_index(const void *fdt __unused,
+static inline TEE_Result gpio_dt_cfg_by_index(const void *fdt __unused,
 					      int nodeoffset __unused,
 					      unsigned int index  __unused,
 					      const char *gpio_name  __unused,
+					      enum gpio_flags mode __unused,
 					      struct gpio **gpio __unused)
 {
 	return TEE_ERROR_NOT_SUPPORTED;
 }
 
+static inline gpio_configure(struct gpio *gpio, enum gpio_flags mode);
```

Comment:
```suggestion
static inline TEE_Result gpio_configure(struct gpio *gpio, enum gpio_flags mode)
{
	return TEE_ERROR_NOT_SUPPORTED;
}
```
CI build test complained on that.

Reply:
Yes the function should be removed as it is already defined independently of CFG_DT flags